src/transformers/trainer.py

```diff
@@ -370,6 +370,19 @@ def __init__(
         else:
             self.is_model_parallel = False
 
+        if (
+            getattr(model, "hf_device_map", None) is not None
+            and len(set(model.hf_device_map.values())) > 1
```

Comment:
Maybe actually check the number of GPUs, cause this could be one GPU and CPU here.

Reply:
Multi-device placement should only be on GPUs for naive pipelining to work, right? Offloading to CPU/disk won't work, isn't it the case?

Reply:
I think offloading to CPU/disk won't work yes, I am also unsure if CPU/disk offload training works out of the box with `accelerate` (without DeepSpeed)